sql/sem/builtins: TestSerialNormalizationWithUniqueUnorderedID failed

sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10914749?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10914749?buildTab=artifacts#/) on master @ [334e261af905918edf3ffe43e07f724e9f9b0b01](https://github.com/cockroachdb/cockroach/commits/334e261af905918edf3ffe43e07f724e9f9b0b01):

Fatal error:

```
panic: test timed out after 4m55s

```
Stack: 

```
goroutine 23148 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
I230714 14:54:34.610793 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r6/1:/Table/{0-3}] 717  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.610885 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r45/1:/Table/4{3-4}] 718  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.610975 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r44/1:/Table/4{2-3}] 719  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611078 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r43/1:/Table/4{1-2}] 720  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611178 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r13/1:/Table/1{1-2}] 721  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611276 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r23/1:/Table/2{1-2}] 722  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611369 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r49/1:/Table/4{7-8}] 723  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611462 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r39/1:/Table/3{7-8}] 724  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611553 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r22/1:/Table/2{0-1}] 725  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611643 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r11/1:/Table/{7-8}] 726  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611733 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r51/1:/Table/5{0-1}] 727  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611832 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r31/1:/{Table/29-NamespaceTab…}] 728  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.611954 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r3/1:/System/{NodeLive…-tsd}] 729  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612058 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r63/1:/Table/{62-106}] 730  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612152 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r35/1:/Table/3{3-4}] 731  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612381 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 732  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612536 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r14/1:/Table/1{2-3}] 733  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612636 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r56/1:/Table/5{5-6}] 734  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612733 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r20/1:/Table/1{8-9}] 735  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612826 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r59/1:/Table/5{8-9}] 736  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.612920 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r25/1:/Table/2{3-4}] 737  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.613016 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r10/1:/Table/{6-7}] 738  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.613111 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r42/1:/Table/4{0-1}] 739  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.613643 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 740  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.613925 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r57/1:/Table/5{6-7}] 741  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.614052 3066 13@kv/kvserver/replicate_queue.go:751  [T1,n1,replicate,s1,r41/1:/Table/{39-40}] 742  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230714 14:54:34.621581 769 2@gossip/gossip.go:552  [T1,n1] 743  gossip status (ok, 1 node)
I230714 14:54:34.621581 769 2@gossip/gossip.go:552  [T1,n1] 743 +gossip client (0/3 cur/max conns)
I230714 14:54:34.621581 769 2@gossip/gossip.go:552  [T1,n1] 743 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I230714 14:54:34.754765 772 2@util/log/event_log.go:32  [T1,n1] 744 ={"Timestamp":1689346474754759790,"EventType":"runtime_stats","MemRSSBytes":496705536,"GoroutineCount":605,"MemStackSysBytes":7143424,"GoAllocBytes":281017376,"GoTotalBytes":336684360,"GoStatsStaleness":0.001397958,"HeapFragmentBytes":14394336,"HeapReservedBytes":8183808,"HeapReleasedBytes":24674304,"CGoAllocBytes":86849984,"CGoTotalBytes":119357440,"CGoCallRate":0.09939432,"CPUUserPercent":115.19801,"CPUSysPercent":0.29818296,"GCPausePercent":0.16196814,"GCRunCount":76,"NetHostRecvBytes":23845822,"NetHostSendBytes":23845822}
I230714 14:54:34.754904 772 2@server/status/runtime_log.go:47  [T1,n1] 745  runtime stats: 474 MiB RSS, 605 goroutines (stacks: 6.8 MiB), 268 MiB/321 MiB Go alloc/total(stale) (heap fragmentation: 14 MiB, heap reserved: 7.8 MiB, heap released: 24 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 115.2/0.3 %(u/s)time, 0.2 %gc (76x), 23 MiB/23 MiB (r/w)net
I230714 14:54:40.804794 20729 kv/kvserver/replica_command.go:439  [T1,n1,split,s1,r79/1:/{Table/106/1/…-Max}] 746  initiating a split of this range at key /Table/106/1/2031255669139308545 [r80] (load at key /Table/106/1/2031255669139308545 (cpu 1.9s, 13.26 batches/sec, 0.10 raft mutations/sec))
I230714 14:54:44.699811 772 2@util/log/event_log.go:32  [T1,n1] 747 ={"Timestamp":1689346484699806039,"EventType":"runtime_stats","MemRSSBytes":466800640,"GoroutineCount":605,"MemStackSysBytes":9830400,"GoAllocBytes":195329704,"GoTotalBytes":338511176,"GoStatsStaleness":0.001239617,"HeapFragmentBytes":23609688,"HeapReservedBytes":83927040,"HeapReleasedBytes":22847488,"CGoAllocBytes":86849984,"CGoTotalBytes":119357440,"CGoCallRate":0.100552574,"CPUUserPercent":88.98903,"CPUSysPercent":0.6033154,"GCRunCount":76,"NetHostRecvBytes":17837962,"NetHostSendBytes":17837962}
I230714 14:54:44.699944 772 2@server/status/runtime_log.go:47  [T1,n1] 748  runtime stats: 445 MiB RSS, 605 goroutines (stacks: 9.4 MiB), 186 MiB/323 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 80 MiB, heap released: 22 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 89.0/0.6 %(u/s)time, 0.0 %gc (76x), 17 MiB/17 MiB (r/w)net
I230714 14:54:45.126833 712 kv/kvserver/queue.go:616  [T1,n1,s1,r5/1:/{Systemtse-Table/0},raft] 749  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I230714 14:54:51.987381 2818 13@kv/kvserver/split/decider.go:271  [T1,n1,s1,r80/1:/{Table/106/1/…-Max}] 750  no split key found: insufficient counters = 0, imbalance = 20, most popular key occurs in 11% of samples
I230714 14:54:54.712744 772 2@util/log/event_log.go:32  [T1,n1] 751 ={"Timestamp":1689346494712739585,"EventType":"runtime_stats","MemRSSBytes":467873792,"GoroutineCount":606,"MemStackSysBytes":11993088,"GoAllocBytes":259840000,"GoTotalBytes":338699592,"GoStatsStaleness":0.003298985,"HeapFragmentBytes":12969984,"HeapReservedBytes":27918336,"HeapReleasedBytes":22724608,"CGoAllocBytes":86849984,"CGoTotalBytes":119357440,"CGoCallRate":0.09987083,"CPUUserPercent":113.45327,"CPUSysPercent":0.69909585,"GCRunCount":76,"NetHostRecvBytes":14234193,"NetHostSendBytes":14234123}
I230714 14:54:54.712899 772 2@server/status/runtime_log.go:47  [T1,n1] 752  runtime stats: 446 MiB RSS, 606 goroutines (stacks: 11 MiB), 248 MiB/323 MiB Go alloc/total(stale) (heap fragmentation: 12 MiB, heap reserved: 27 MiB, heap released: 22 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 113.5/0.7 %(u/s)time, 0.0 %gc (76x), 14 MiB/14 MiB (r/w)net
I230714 14:54:54.866710 21344 kv/kvserver/replica_command.go:439  [T1,n1,split,s1,r80/1:/{Table/106/1/…-Max}] 753  initiating a split of this range at key /Table/106/1/2115698162152505345 [r81] (load at key /Table/106/1/2115698162152505345 (cpu 2s, 12.99 batches/sec, 0.09 raft mutations/sec))
I230714 14:54:59.125495 728 kv/kvserver/queue.go:616  [T1,n1,s1,r30/1:/Table/2{8-9},raft] 754  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I230714 14:55:04.722788 772 2@util/log/event_log.go:32  [T1,n1] 755 ={"Timestamp":1689346504722781405,"EventType":"runtime_stats","MemRSSBytes":462012416,"GoroutineCount":605,"MemStackSysBytes":9732096,"GoAllocBytes":201851480,"GoTotalBytes":330704200,"GoStatsStaleness":0.001327697,"HeapFragmentBytes":21093800,"HeapReservedBytes":72146944,"HeapReleasedBytes":30720000,"CGoAllocBytes":86849984,"CGoTotalBytes":119357440,"CGoCallRate":0.09989969,"CPUUserPercent":119.38013,"CPUSysPercent":0.5993981,"GCPausePercent":0.25646996,"GCRunCount":77,"NetHostRecvBytes":698596,"NetHostSendBytes":698596}
I230714 14:55:04.722937 772 2@server/status/runtime_log.go:47  [T1,n1] 756  runtime stats: 441 MiB RSS, 605 goroutines (stacks: 9.3 MiB), 192 MiB/315 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 69 MiB, heap released: 29 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 119.4/0.6 %(u/s)time, 0.3 %gc (77x), 682 KiB/682 KiB (r/w)net
I230714 14:55:05.964903 11573 13@kv/kvserver/split/decider.go:271  [T1,n1,s1,r81/1:/{Table/106/1/…-Max}] 757  no split key found: insufficient counters = 1, imbalance = 19, most popular key occurs in 12% of samples
I230714 14:55:10.933247 15932 kv/kvserver/replica_command.go:439  [T1,n1,split,s1,r81/1:/{Table/106/1/…-Max}] 758  initiating a split of this range at key /Table/106/1/2239969364370259969 [r82] (load at key /Table/106/1/2239969364370259969 (cpu 1.9s, 15.58 batches/sec, 0.09 raft mutations/sec))
I230714 14:55:14.709777 772 2@util/log/event_log.go:32  [T1,n1] 759 ={"Timestamp":1689346514709771103,"EventType":"runtime_stats","MemRSSBytes":436539392,"GoroutineCount":605,"MemStackSysBytes":7274496,"GoAllocBytes":143933312,"GoTotalBytes":303908168,"GoStatsStaleness":0.001343227,"HeapFragmentBytes":44744832,"HeapReservedBytes":81944576,"HeapReleasedBytes":57647104,"CGoAllocBytes":86849952,"CGoTotalBytes":119357440,"CGoCallRate":0.100130275,"CPUUserPercent":175.82875,"CPUSysPercent":0.9011724,"GCPausePercent":0.0016358683,"GCRunCount":78,"NetHostRecvBytes":22792290,"NetHostSendBytes":22792290}
I230714 14:55:14.709912 772 2@server/status/runtime_log.go:47  [T1,n1] 760  runtime stats: 416 MiB RSS, 605 goroutines (stacks: 6.9 MiB), 137 MiB/290 MiB Go alloc/total(stale) (heap fragmentation: 43 MiB, heap reserved: 78 MiB, heap released: 55 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 175.8/0.9 %(u/s)time, 0.0 %gc (78x), 22 MiB/22 MiB (r/w)net
I230714 14:55:20.124788 747 kv/kvserver/queue.go:616  [T1,n1,s1,r71/1:/Table/106/1/1{23890…-33516…},raft] 761  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I230714 14:55:21.986332 22818 kv/kvserver/replica_command.go:439  [T1,n1,split,s1,r82/1:/{Table/106/1/…-Max}] 762  initiating a split of this range at key /Table/106/1/2324834069848522753 [r83] (load at key /Table/106/1/2324834069848522753 (cpu 1.9s, 14.54 batches/sec, 0.08 raft mutations/sec))
I230714 14:55:24.725506 772 2@util/log/event_log.go:32  [T1,n1] 763 ={"Timestamp":1689346524725500884,"EventType":"runtime_stats","MemRSSBytes":437600256,"GoroutineCount":605,"MemStackSysBytes":9764864,"GoAllocBytes":209501080,"GoTotalBytes":303908168,"GoStatsStaleness":0.00134913,"HeapFragmentBytes":21660776,"HeapReservedBytes":36970496,"HeapReleasedBytes":57647104,"CGoAllocBytes":86878624,"CGoTotalBytes":119484416,"CGoCallRate":0.09984294,"CPUUserPercent":121.808395,"CPUSysPercent":0.29952884,"GCRunCount":78,"NetHostRecvBytes":18100121,"NetHostSendBytes":18100121}
I230714 14:55:24.725630 772 2@server/status/runtime_log.go:47  [T1,n1] 764  runtime stats: 417 MiB RSS, 605 goroutines (stacks: 9.3 MiB), 200 MiB/290 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 35 MiB, heap released: 55 MiB), 83 MiB/114 MiB CGO alloc/total (0.1 CGO/sec), 121.8/0.3 %(u/s)time, 0.0 %gc (78x), 17 MiB/17 MiB (r/w)net
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSerialNormalizationWithUniqueUnorderedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29733

--- a/pkg/sql/sem/builtins/builtins_test.go
+++ b/pkg/sql/sem/builtins/builtins_test.go
@@ -31,6 +31,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
@@ -100,6 +101,7 @@ func TestMapToUniqueUnorderedID(t *testing.T) {
 // by insertions guarantees a (somewhat) uniform distribution of the data.
 func TestSerialNormalizationWithUniqueUnorderedID(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t, "the test is too slow and the goodness of fit test "+
 		"assumes large N")
@@ -109,7 +111,15 @@ func TestSerialNormalizationWithUniqueUnorderedID(t *testing.T) {
 	// This test can flake when the random data is not distributed evenly.
 	err := retry.WithMaxAttempts(ctx, retry.Options{}, attempts, func() error {
 
-		params := base.TestServerArgs{}
+		params := base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				SQLEvalContext: &eval.TestingKnobs{
+					// We disable the randomization of some batch sizes because
+					// with some low values the test takes much longer.
+					ForceProductionValues: true,
+				},
+			},
+		}
 		s, db, _ := serverutils.StartServer(t, params)
 		defer s.Stopper().Stop(ctx)
 

